### PR TITLE
feat: refactor registration form

### PR DESCRIPTION
**Pull request by zainab-amir**
#### Context
- The default registration fields are covered in `RegistrationPage.jsx`. These fields are:
   - name
   - email
   - username
   - password
 - Persist state for errors and field values on tab switch
- The configurable registration fields i.e honor code, terms of service, city, profession etc are covered in `ConfigurableRegistrationForm.jsx`.
- For edX only `SHOW_CONFIGURABLE_EDX_FIELDS` is set to true. By setting it to true, we show the country and honor code fields by default even if `ENABLE_DYNAMIC_REGISTRATION_FIELDS` is false.

#### Tests:
- Added new tests to cover updated code.
- Removed tests that were no longer valid.

### CODE REVIEW

**Comment by arbrandes** [comment]
@zainab-amir, is this the refactoring we were discussing in [the Olive Matrix](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3571056752/Olive+MFE+Candidate+Requirements+Matrix?focusedCommentId=3590684686)?

File: src/register/registrationFields/CountryField.jsx [hunk 1]
```diff
@@ -20,15 +20,15 @@ const CountryField = (props) => {
   const [displayValue, setDisplayValue] = useState('');
   const [trailingIcon, setTrailingIcon] = useState(null);
 
-  const onBlurHandler = (event, itemClicked = false) => {
+  const onBlurHandler = (event, itemClicked = false, countryClicked = '') => {
```
Comment (mubbsharanwar) [inline src/register/registrationFields/CountryField.jsx:23]: nit: `countryClicked ` to `countryName`

File: src/register/tests/RegistrationPage.test.jsx [hunk 2]
```diff
@@ -938,6 +945,27 @@ describe('RegistrationPage', () => {
       expect(registrationPage.find('input#password').props().value).toEqual('password1');
       expect(registrationPage.find('.email-warning-alert-link').first().text()).toEqual('john.doe@hotmail.com');
     });
+
+    it('should set country in component state on country change with translations', () => {
```
Comment (mubbsharanwar) [inline src/register/tests/RegistrationPage.test.jsx:949]: nit: should set the component country state on onChange while the page is translated.

File: src/register/tests/RegistrationPage.test.jsx [hunk 3]
```diff
@@ -938,6 +945,27 @@ describe('RegistrationPage', () => {
...
+      getLocale.mockImplementation(() => ('ar-ae'));
+
+      const registrationPage = mount(reduxWrapper(<IntlRegistrationPage {...props} />));
+      registrationPage.find('input#country').simulate('focus');
+      registrationPage.find('button.dropdown-item').at(0).simulate('click', { target: { value: 'أفغانستان ', name: 'countryItem' } });
+      expect(registrationPage.find('div[feedback-for="country"]').exists()).toBeFalsy();
+    });
+
+    it('should set country in component state on country change with chrome translations', () => {
```
Comment (mubbsharanwar) [inline src/register/tests/RegistrationPage.test.jsx:958]: should set the component country state on onChange with the chrome translations

**Review by mubbsharanwar — APPROVED** [review]

**Pull request merged**